optaweb-vehicle-routing-backend/src/main/java/org/optaweb/vehiclerouting/plugin/planner/SolutionPublisher.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaweb.vehiclerouting.plugin.planner;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.optaplanner.examples.vehiclerouting.domain.Customer;
+import org.optaplanner.examples.vehiclerouting.domain.Depot;
+import org.optaplanner.examples.vehiclerouting.domain.Vehicle;
+import org.optaplanner.examples.vehiclerouting.domain.VehicleRoutingSolution;
+import org.optaweb.vehiclerouting.service.route.RouteChangedEvent;
+import org.optaweb.vehiclerouting.service.route.ShallowRoute;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.stereotype.Component;
+
+/**
+ * Converts planning solution to a {@link RouteChangedEvent} and publishes it so that it can be processed by other
+ * components that listen for this type of event.
+ */
+@Component
+class SolutionPublisher {
+
+    private static final Logger logger = LoggerFactory.getLogger(SolutionPublisher.class);
+
+    private final ApplicationEventPublisher eventPublisher;
+
+    SolutionPublisher(ApplicationEventPublisher eventPublisher) {
```

Comment:
Good question. Answer: no. Explanation: https://github.com/spring-projects/spring-framework/blob/v5.2.0.RC1/spring-beans/src/main/java/org/springframework/beans/factory/annotation/AutowiredAnnotationBeanPostProcessor.java#L90-L92.

However this raises another question. How to prevent your question from occurring again? Probably we don't want to add `// @Autowired not needed here` above all bean constructors. I'm starting to think that the best approach is to put `@Autowired` everywhere although it's not required. Yes, I can be smart (lazy) and not use it when it's not needed but that's not friendly towards other people reading the code. Correct?

Reply:
Thanks for the link, I was not aware of that. Question is how broad a problem it is. Spring usually follows "convention over configuration".
The comment does not sound like a solution; we either put the annotation there or omit it.